dialvia/http.go

```diff
@@ -118,8 +118,10 @@ func (d *HTTPProxyDialer) DialContextR(ctx context.Context, network, addr string
 
 	// Don't send the default Go HTTP client User-Agent.
 	req.Header.Add("User-Agent", "")
-	if d.proxyURL.User != nil {
-		req.Header.Add("Proxy-Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(d.proxyURL.User.String())))
+	if u := d.proxyURL.User; u != nil {
+		pass, _ := u.Password()
+		auth := u.Username() + ":" + pass
```

Comment:
This is wrong. If password is empty it should be just `<username>`.

Reply:
Can change it, tho.

Reply:
That's actually a legit question if no password is the same as empty password.
I guess it depends.

Reply:
It is the same behaviour as for normal requests https://github.com/saucelabs/forwarder/blob/09bd2ad6723ef8f53bb8a98764309aa988e22147/http_proxy.go#L510

```go
func (hp *HTTPProxy) setBasicAuth(req *http.Request) error {
	if req.Header.Get("Authorization") == "" {
		if u := hp.creds.MatchURL(req.URL); u != nil {
			p, _ := u.Password()
			req.SetBasicAuth(u.Username(), p)
		}
	}

	return nil
}

func (r *Request) SetBasicAuth(username, password string) {
	r.Header.Set("Authorization", "Basic "+basicAuth(username, password))
}

func basicAuth(username, password string) string {
	auth := username + ":" + password
	return base64.StdEncoding.EncodeToString([]byte(auth))
}
```

I'm not sure if it is intended to send basic-auth without password.